### PR TITLE
Uldaman

### PR DESCRIPTION
**Pull request by sipertruk**
- Ironaya targets immediately whoever activated the door when it opens.
- Crystalline Slumber (basilisk sleep) has a cast time.
- The first altar event fix, it was automatically started and blocked the door upon failing (there was 2 concurrent implementations).
- Altar summoning needs 3 players in vanilla.
- Altar events proper resets, upon failling everything respawn at original location.
- On Archaedas death all his still standing minions despawn.
- Earthen/Stone Guardians/Archaedas proper awakening spells with accurate visual.
- Archaedas resets when pulled out of his room. Because it was(is) considered as bug to be able to pull boss through closed doors.
- Spells 10254 and 10255 need to be used separately because we need different visuals, see the SpellEffect.cpp/SpellAura.cpp modifications.
- Earthen Custodians use Reconstruct on Archaedas (Health transfer).
- Altars are animated.
- Lore Keeper of Norgannon's faction fix (none).
- Obsidian Sentinel uses the spell Splintered Obsidian (size modification).

Related bug report :
https://elysium-project.org/bugtracker/issue/733
https://elysium-project.org/bugtracker/issue/3445 ? (Is it fixing this ? I don't have the right to see the content)

The visual spell for altars rituals looks wrong, I could get it right with a hack in Spell.cpp but I suppose the true problem is deeper in gameobject class.

It's my first work ever on mangos, if anything is out of place please tell, I want to learn.

### CODE REVIEW

File: src/scripts/eastern_kingdoms/uldaman/boss_archaedas.cpp [hunk 1]
```diff
@@ -321,79 +394,111 @@ CreatureAI* GetAI_mob_archaedas_minions(Creature* creature)
...
-
-    void Aggro(Unit* pWho)
+    
+    uint32 CountSummoners()
     {
-        me->setFaction(14);
-        me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE | UNIT_FLAG_DISABLE_MOVE);
+        const Map::PlayerList& players = me->GetMap()->GetPlayers();
+        uint32 count = 0;
+        for (Map::PlayerList::const_iterator it = players.begin(); it != players.end(); ++it)
```
Comment (Zerixx) [inline src/scripts/eastern_kingdoms/uldaman/boss_archaedas.cpp]: Now, it's not completely necessary, but range based for loops for (auto const &it : players) OR for (Map::PlayerList const &it : players) will clean up this, and all other iterative code that fits this general setup as well. 

Like I said, not necessary, but there's nothing wrong with simplifying code where applicable. Good work, nevertheless.

Comment (sipertruk) [inline src/scripts/eastern_kingdoms/uldaman/boss_archaedas.cpp]: Is Map::PlayerList compatible with that ? I can't get it to work, using gcc-6.

Comment (sipertruk) [inline src/scripts/eastern_kingdoms/uldaman/boss_archaedas.cpp]: But it works for std::vectors. Thanks.

File: sql/migrations/20170331165457_world.sql [hunk 2]
```diff
@@ -7,3 +7,9 @@ INSERT INTO `spell_script_target` (`entry`, `type`, `targetEntry`) VALUES ('1025
 INSERT INTO `spell_script_target` (`entry`, `type`, `targetEntry`) VALUES ('10259', '1', '7309');
 -- Lore Keeper of Norgannon is not in Cenarion Circle
 UPDATE `creature_template` SET `faction_A`='35', `faction_H`='35' WHERE `entry`='7172';
+-- Obsidian Sentinel second action on hp events is not happening, it hould be triggered
+UPDATE `elysium_world`.`creature_ai_scripts` SET `action2_param3`='2' WHERE `id`='702301';
```
Comment (brotalnia) [inline sql/migrations/20170331165457_world.sql]: Remove the `elysium_world` part. Not everyone has the same name for the database.

**Comment by sipertruk** [comment]
I don't see anything else to fix.

**Pull request merged**